File: web/src/api/user-asset.ts
import apiClient from './client'

export interface UserAssetSummary {
  points: number
  balance: number
  subscription: string
  member_level: string
}

export interface TransactionRecord {
  id: string
  type: string
  method: string
  amount: number
  status: string
  created_at: string
}

export interface PointRecord {
  id: number
  type: string
  change: number
  balance: number
  created_at: string
}

export interface UsageStats {
  operator_calls: number
  ai_model_calls: number
  token_usage: number
}

export const userAssetApi = {
  getSummary() {
    return apiClient.get<UserAssetSummary>('/user/assets/summary')
  },
  getTransactions(params?: { limit?: number; offset?: number }) {
    return apiClient.get<TransactionRecord[]>('/user/assets/transactions', { params })
  },
  getPoints(params?: { limit?: number; offset?: number }) {
    return apiClient.get<PointRecord[]>('/user/assets/points', { params })
  },
  getUsage() {
    return apiClient.get<UsageStats>('/user/assets/usage')
  },
  recharge(data: { amount: number; channel: string }) {
    return apiClient.post<{ order_no: string; pay_url?: string; qrcode?: string }>('/user/assets/recharge', data)
  },
  checkIn() {
    return apiClient.post<{ message: string }>('/user/assets/checkin')
  },
  subscribe(planName: string) {
    return apiClient.post<{ message: string }>('/user/assets/subscribe', { plan_name: planName })
  }
}

File: internal/api/dto/user_asset.go
package dto

import "time"

// UserAssetSummary 综合资产概览
type UserAssetSummary struct {
	Points       int64   `json:"points"`
	Balance      float64 `json:"balance"`
	Subscription string  `json:"subscription"`
	MemberLevel  string  `json:"member_level"`
}

// PaymentMethod 支付方式
type PaymentMethod struct {
	ID          string `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	Icon        string `json:"icon"`
}

// TransactionRecord 交易记录
type TransactionRecord struct {
	ID        string    `json:"id"`
	Type      string    `json:"type"`
	Method    string    `json:"method"`
	Amount    float64   `json:"amount"`
	Status    string    `json:"status"`
	CreatedAt time.Time `json:"created_at"`
}

// PointRecord 积分记录
type PointRecord struct {
	ID        int64     `json:"id"`
	Type      string    `json:"type"`
	Change    int64     `json:"change"`
	Balance   int64     `json:"balance"`
	CreatedAt time.Time `json:"created_at"`
}

// SubscriptionPlan 订阅计划
type SubscriptionPlan struct {
	Name     string   `json:"name"`
	Price    float64  `json:"price"`
	Features []string `json:"features"`
	IsActive bool     `json:"is_active"`
}

// UsageStats 使用统计
type UsageStats struct {
	OperatorCalls int64 `json:"operator_calls"`
	AIModelCalls  int64 `json:"ai_model_calls"`
	TokenUsage    int64 `json:"token_usage"` // 单位: Token
}
